pkg/route/graph/analysis/analysis.go

```diff
@@ -0,0 +1,46 @@
+package analysis
+
+import (
+	"fmt"
+
+	"github.com/gonum/graph"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	kubeedges "github.com/openshift/origin/pkg/api/kubegraph"
+	kubegraph "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+	routegraph "github.com/openshift/origin/pkg/route/graph/nodes"
+)
+
+const (
+	MissingRoutePortWarning = "MissingRoutePort"
+)
+
+// FindMissingPortMapping checks all routes and reports those that don't have a route port specified
+func FindMissingPortMapping(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+route:
+	for _, uncastRouteNode := range g.NodesByKind(routegraph.RouteNodeKind) {
+		for _, uncastServiceNode := range g.SuccessorNodesByEdgeKind(uncastRouteNode, kubeedges.ExposedThroughServiceEdgeKind) {
```

Comment:
> This doesn't look right. The error is based on whether the route has a target port. I don't think the service is a related object. It wasn't involved in the decision at all.

Reading through the doc, it looks like this is only something we need to call out if the services themselves have more than one port exposed?  That would be a reason for having services be related.  Add the test to see if its actually a problem and then be more descriptive in the message.